src/midgard/point2.cc

```diff
@@ -47,8 +47,9 @@ PointXY<PrecisionT>::ClosestPoint(const std::vector<PointXY<PrecisionT>>& pts) c
     const PointXY<PrecisionT>& p1 = pts[index + 1];
 
     // Construct vector v1 - represents the segment.  Skip 0 length segments
+    // that are not at the end of the line.
     v1.Set(p0, p1);
-    if (v1.x() == 0.0f && v1.y() == 0.0f) {
+    if (v1.x() == 0.0f && v1.y() == 0.0f && index < pts.size() - 2) {
```

Comment:
It's worth mentioning also that this change shouldn't break the algorithm. When we hit a 0 length segment at the end of the polyline and continue through the for loop, we know that `v1` is the zero vector. So on line 62, `dot` will equal 0. This causes the if statement starting on line 63 to have a true condition, and the things done in that if statement are what the algorithm should already be doing in this case.